### ISSUE TITLE
Fix Request: False positive for AvoidMutableList

### ISSUE THREAD

**Issue by rodrigolopes** [jborgers/PMD-jPinpoint-rules#288]
Collections.emptyList() is unmodifiable. 

![5dd788f6-4eec-4735-93a4-0c657692b4fa](https://github.com/jborgers/PMD-jPinpoint-rules/assets/570477/9f1a654c-e445-42a8-92c9-a15b7de30d8c)

**Comment by jborgers** [comment]
For this piece of code, I cannot reproduce the violation. 
However, if I add a constructor like: 
    ```
public ClientCollection(List<Client> clients) {
        this.clients = clients;
    }
```

There is a violation, which is correct when the this.clients list is not actually modified in the class.
The location of the violation is on the field, which would be better located on the assignment. Yet, since this rule is combination of several cases, that is hard to achieve. Improving the description/documentation and adding this case as example is expected to be sufficient to not be experienced as false positive.
